app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -12,15 +12,19 @@ import com.spotify.android.appremote.api.SpotifyAppRemote
 import com.spotify.protocol.client.Subscription
 import com.spotify.protocol.types.*
 import kotlinx.coroutines.*
+import kotlinx.serialization.decodeFromString
```

Comment:
The app already has Gson included, which can do class de/serialization, what does the Kotlin version do differently?

Reply:
Originally I had used the Kotlinx serialization due to my initial implementation dealing with default parameters for the LikedSongsState and Gson didn't play nicely with those for some reason but now with the latest implementation there really is no need for kotlinx serialization anymore and Gson will do nicely, updated the implementation to use Gson and removed kotlinx serialization